R/calc_views.R

```diff
@@ -61,6 +62,13 @@ group_views <- function(gsplot){
   return(append(views, non.views))
 }
 
+append_replace <- function(old.list, new.list){
+  out.list <- old.list
+  out.list[names(old.list) %in% names(new.list)] <- new.list[names(out.list[names(old.list) %in% names(new.list)])]
+  new.list[names(new.list) %in% names(old.list)] <- NULL
+  out.list <- append(out.list, new.list)
+  return(out.list)
+}
```

Comment:
I think i need to keep `set_window` in there, but we should be able to get rid of all or most of it in the future. I just don't really have a good grasp of what it does, and playing around w/ it breaks things.